packages/modules/e3dc/bat.py

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python3
+import logging
+
+from modules.common import modbus
+from modules.common.component_state import BatState
+from modules.common.component_type import ComponentDescriptor
+from modules.common.modbus import ModbusDataType, Endian
+from modules.common.fault_state import ComponentInfo
+from modules.common.store import get_bat_value_store
+from modules.common.simcount._simcounter import SimCounter
+from modules.e3dc.config import E3dcBatSetup
+
+
+log = logging.getLogger(__name__)
+
+
+def read_bat(client: modbus.ModbusTcpClient_) -> [int, int]:
+    # 40082 SoC
+    soc = client.read_holding_registers(40082, ModbusDataType.INT_16, unit=1)
+    # 40069 Speicherleistung
+    power = client.read_holding_registers(40069, ModbusDataType.INT_32, wordorder=Endian.Little, unit=1)
+    return soc, power
+
+
+class E3dcBat:
+    def __init__(self,
+                 device_id: int,
+                 component_config: E3dcBatSetup) -> None:
+        self.component_config = component_config
+        self.__device_id = device_id
+        # bat
+        self.sim_counter = SimCounter(self.__device_id, self.component_config.id, prefix="speicher")
```

Comment:
```suggestion
        # bat
        self.sim_counter = SimCounter(device_id, self.component_config.id, prefix="speicher")
```
Da die Eigenschaft `__device_id` sonst nie gelesen wird könntest du die auch einsparen.

Reply:
geändert